plugins/discourse-narrative-bot/plugin.rb

```diff
@@ -114,7 +114,7 @@ module ::DiscourseNarrativeBot
     notification = Notification.where(topic_id: topic.id, post_number: first_post.post_number).first
     if notification.present?
       Notification.read(self, notification.id)
-      self.saw_notification_id(notification.id)
+      self.bump_last_seen_notification!
```

Comment:
I think we need to wrap this line in an if condition if we want this to be 100% equivalent to `saw_notification_id`. The if condition would need to check that `user.notifications.visible.maximum(:id) == notification.id` and only call `bump_last_seen_notification!` if the condition is true. Without this condition, we could end up in a situation where the user misses a notification that was created after the notification that we're trying to mark as read here because `bump_last_seen_notification!` marks the most recent notification as seen instead of targeting a specific notification ID.

Or, I think we can drop this line entirely and just get away with the `Notification.read` call above this line which will be enough to dismiss the notification.